core/services/ocr2/plugins/vault/transmitter.go

```diff
@@ -0,0 +1,66 @@
+package vault
+
+import (
+	"context"
+	"encoding/binary"
+	"fmt"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/actions/vault"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/consensus/requests"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/libocr/offchainreporting2/types"
+	"github.com/smartcontractkit/libocr/offchainreporting2plus/ocr3types"
+	"google.golang.org/protobuf/proto"
+)
+
+type Transmitter struct {
+	lggr        logger.Logger
+	store       *requests.Store[*Request]
+	fromAccount types.Account
+}
+
+func NewTransmitter(lggr logger.Logger, fromAccount types.Account, store *requests.Store[*Request]) *Transmitter {
+	return &Transmitter{
+		lggr:        lggr.Named("VaultTransmitter"),
+		store:       store,
+		fromAccount: fromAccount,
+	}
+}
+
+func (c *Transmitter) Transmit(ctx context.Context, cd types.ConfigDigest, seqNr uint64, rwi ocr3types.ReportWithInfo[[]byte], sigs []types.AttributedOnchainSignature) error {
+	info := &vault.ReportInfo{}
+	err := proto.Unmarshal(rwi.Info, info)
+	if err != nil {
+		return err
+	}
+
+	req := c.store.Get(info.Id)
+	if req == nil {
+		return fmt.Errorf("request with ID %s not found", info.Id)
+	}
+
+	seqToEpoch := make([]byte, 32)
+	binary.BigEndian.PutUint32(seqToEpoch[32-5:32-1], uint32(seqNr))
+	zeros := make([]byte, 32)
+	responseCtx := append(append(cd[:], seqToEpoch[:]...), zeros...)
+
+	var signatures [][]byte
```

Comment:
Was your intention to drop signatures from the report, in case request was GetSecrets?
You could technical check for info.RequestType here, and drop signatures for GetSecrets type.